snapshot: fix terminology inconsistency

AkihiroSuda opened the pull request:
I think the terminology should be consistent in Phase One.

This PR replaces `LayerManipulator` and `SnapshotManipulator`  with `SnapshotManager`.
Please let me know if the opposite direction is correct.

Signed-off-by: Akihiro Suda <suda.akihiro@lab.ntt.co.jp>

stevvooe commented on design/snapshots.md:
```diff
@@ -142,7 +142,7 @@ that applies the contents of the layer to target location and calculates the
 implementation):
 
-	digest, err := unpackLayer(tmpLocation, layer) // unpack into layer location
+	digest, err := unpackLayer(tmpLocation, layerPath) // unpack into layer location
```
This was correct. The `layer` here is the `io.Reader` of the actual layer content.

stevvooe commented:
@AkihiroSuda As of my latest work, the lower level drivers will be known as "Snapshot Drivers". The "Snapshot Manager" will likely be a component either comprising a direct interface to the drivers or something with more common methods. In that scenario, `SnapshotManager` (the type, not the concept) will be a concrete type that takes a driver as the argument.

AkihiroSuda commented:
@stevvooe 
Thank you for clarifying.

Updated PR

 * Removed all "sm" from the driver packages (`overlay`, `btrfs`, and `naive`), because "Snapshot Drivers" are not "Snapshot Manager"
 * Renamed `Overlayfs` to `Overlay`, because the package name is `overlay`
 * Added `layer, err := os.Open(layerPath)` to the design doc, so as to clarify what is `layer`.

AkihiroSuda commented:
rebased to master

stevvooe commented:
LGTM

@AkihiroSuda Thanks!
